source/simulator/helper_functions.cc

```diff
@@ -1784,34 +1789,36 @@ namespace aspect
 
           material_model->evaluate(in, out);
 
-          // interpolate material properties onto the compositional fields
+          // Interpolate material properties onto the compositional fields
           for (unsigned int j=0; j<dof_handler.get_fe().base_element(introspection.base_elements.compositional_fields).dofs_per_cell; ++j)
             {
               const unsigned int composition_idx
                 = dof_handler.get_fe().component_to_system_index(introspection.component_indices.compositional_fields[c],
                                                                  /*dof index within component=*/ j);
 
-              // skip entries that are not locally owned:
+              // Skip degrees of freedom that are not locally owned. These
+              // will eventually be handled by one of the other processors.
               if (dof_handler.locally_owned_dofs().is_element(local_dof_indices[composition_idx]))
                 {
                   Assert(numbers::is_finite(prescribed_field_out->prescribed_field_outputs[j][c]),
                          ExcMessage("You are trying to use a prescribed advection field, "
                                     "but the material model you use does not fill the PrescribedFieldOutputs "
                                     "for your prescribed field, which is required for this method."));
 
-                  distributed_vector(local_dof_indices[composition_idx]) = prescribed_field_out->prescribed_field_outputs[j][c];
+                  distributed_vector(local_dof_indices[composition_idx])
+                    = prescribed_field_out->prescribed_field_outputs[j][c];
                 }
             }
         }
 
-    // put the final values into the solution vector
-
+    // Put the final values into the solution vector, also
+    // updating the ghost elements of the 'solution' vector.
     const unsigned int block_c = introspection.block_indices.compositional_fields[c];
     distributed_vector.block(block_c).compress(VectorOperation::insert);
     solution.block(block_c) = distributed_vector.block(block_c);
 
     // We also want to copy the values into the old solution, because it might
-    // be used in other parts of the code
+    // be used in other parts of the code.
```

Comment:
Wait, did I just manage to win our 5 year long battle on punctuation in comments? :-D

Reply:
@tjhei, I am not so sure. Lines 1746 and 1792 do not have the proper punctuation :grin: 
But we should at least try to be consistent...

Reply:
Yes, you did :-) But that already happened several weeks ago :-))

@jdannberg  We are all inconsistent anyway. Do you want me to change other places as well?

Reply:
It's not really important to me. 
But if we decide that from now on, we always want to have comments starting capitalized and using punctuation, then I would write all of my comments like this too, and if we just keep on doing it long enough, at some point most comments should be written that way. So I guess what I am saying is, we should at least be consistent in all new comments, but I don't care enough about it to delay merging a pull request (there are more useful things that can be done with that time).

Reply:
I agree and appreciate you not holding this PR up over it.

I have for 20 years written comments starting with lower-case letters and no final period. But @tjhei has berated me for a sufficiently long time that I'm willing to change my habits going forward. I have no real preference for one way or the other, but he seems to have.

Reply:
Makes sense to me. For me, text is just easier to read with punctuation. 

So from my side this is ready to merge.